packages/rpc-provider/src/substrate-connect/index.ts

```diff
@@ -154,9 +154,11 @@ export class ScProvider implements ProviderInterface {
       callback?.(decodedResponse);
     };
 
-    const addChain = this.#wellKnownChains.has(this.#spec as ScType.WellKnownChain)
-      ? client.addWellKnownChain
-      : client.addChain;
+    const addChain = this.#sharedSandbox
+      ? (async (...args) => (await this.#sharedSandbox!.#chain)!.addChain(...args)) as ScType.AddChain
+      : this.#wellKnownChains.has(this.#spec as ScType.WellKnownChain)
+        ? client.addWellKnownChain
+        : client.addChain;
```

Comment:
The `this.#sharedSandbox!.#chain` is not transpiled properly see
<img width="718" alt="image" src="https://github.com/polkadot-js/api/assets/1209171/2b4df8ce-b68e-45cb-a43c-1e178ab5658c">

I think this is related to this `polkadot-js/dev` script
https://github.com/polkadot-js/dev/blob/1d7db82109f666dd013d5db04ef2bd6a45f4d74a/packages/dev/scripts/polkadot-dev-build-ts.mjs#L74-L84

Reply:
as an alternative that is coupled to a `polkadot-dev-build-ts.mjs` implementation detail, this can be re-written to
```ts
    const addChain = this.#sharedSandbox
      ? (async (...args) => {
          const source = this.#sharedSandbox!;
          return (await source.#chain)!.addChain(...args);
        }) as ScType.AddChain
      : this.#wellKnownChains.has(this.#spec as ScType.WellKnownChain)
        ? client.addWellKnownChain
        : client.addChain;
```
and it will transpile to
```js
        const addChain = this.__internal__sharedSandbox
            ? (async (...args) => {
                const source = this.__internal__sharedSandbox;
                return (await source.__internal__chain).addChain(...args);
            })
            : this.__internal__wellKnownChains.has(this.__internal__spec)
                ? client.addWellKnownChain
                : client.addChain;
```

Reply:
So the dev script implementation can actually be dropped (after we dropped Node 16 last year). The issue is that TSC compiled everything `#<id>` to WeakMaps for Node 16 compat and it had severe implications for performance. So at best it is/was a horrible workaround.

So the adjustment will be dropped _sometime_, but no idea when that _sometime_ will be - needs some checks all over when done.

Reply:
thx for the info @jacogr 

I'll re-write it as suggested previously so it's more friendly with this script

Reply:
re-written in https://github.com/polkadot-js/api/pull/5776/commits/7df6a77ecee56e57a9fc019a167c201999c14b3a